### DISCUSSION TITLE
Deploy .Net Aspire project to an Azure VM

### DISCUSSION THREAD

**Discussion by junaidahmed92** [microsoft/aspire#5116] [Q&A] [answered]
I am doing some research on .NET Aspire and have a few questions regarding it:

1. I'm working on a microservice architecture with each microservice having its own solution. Can I create a separate Aspire solution and reference all the microservices? Is it recommended? Because in every tutorial I saw, there is a single solution containing multiple microservices.
2. I've deployed all my services on 3 different Azure VMs. Can I deploy an Aspire project to a VM? If yes, I guess I would need an Aspire project on each VM to manage the services of a specific VM. Everywhere I see that the Aspire project is deployed on a container app, and it manages the microservices just like Kubernetes pods. Is there a way to have a centralized Aspire project deployed on a VM that manages all the services?
3. If I deploy my Aspire app to a container app, can it communicate with the microservices deployed on the VM?

**Comment by davidfowl** [comment, marked as answer]
Deploy to an Azure VM using what technology? Aspire doesn't orchestrate your application in production, but it can be used to emit deployment assets. If you are using a VM then you need to decide how to deploy there and then see if aspire can generate assets for that specific technology.
